resources/charts/kubelet/templates/kubelet-ds.yaml

```diff
@@ -0,0 +1,137 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kubelet
+  namespace: kube-system
+  labels:
+    tier: node
+    k8s-app: kubelet
+spec:
+  selector:
+    matchLabels:
+      tier: node
+      k8s-app: kubelet
+  template:
+    metadata:
+      labels:
+        tier: node
+        k8s-app: kubelet
+    spec:
+      containers:
+      - name: kubelet
+        image: {{ .Values.image }}
+        command:
+        - /bin/sh
+        - -c
+        # File /etc/kubernetes/kubelet.env on each host has the `taints` and `labels` that bootstrap
+        # kubelet uses. This mechanism here makes sure that the same labels and taints are used in
+        # this kubelet as well.
+        args:
+        - |
+          echo "./hyperkube kubelet \
+          --node-ip=$(HOST_IP)  \
+          --anonymous-auth=false \
+          --authentication-token-webhook \
+          --authorization-mode=Webhook \
+          --client-ca-file=/etc/kubernetes/ca.crt \
+          --cluster_dns={{ .Values.clusterDNS }} \
+          --cluster_domain={{ .Values.clusterDomain }} \
+          --cni-conf-dir=/etc/kubernetes/cni/net.d \
+          --config=/etc/kubernetes/kubelet.config \
+          --kubeconfig=/etc/kubernetes/kubeconfig \
+          --lock-file=/var/run/lock/kubelet.lock \
+          --network-plugin=cni \
+          --pod-manifest-path=/etc/kubernetes/manifests \
+          --read-only-port=0 \
+          --volume-plugin-dir=/var/lib/kubelet/volumeplugins \
+          --node-labels=$(grep NODE_LABELS /etc/kubernetes/kubelet.env | cut -d'"' -f2) \
+          --register-with-taints=$(grep NODE_TAINTS /etc/kubernetes/kubelet.env | cut -d'"' -f2)
+          " | tee /tmp/start-kubelet.sh &&
+          chmod +x /tmp/start-kubelet.sh &&
+          /tmp/start-kubelet.sh
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/lib/cni
+          name: coreos-var-lib-cni
+          readOnly: false
+        - mountPath: /var/lib/calico
+          name: coreos-var-lib-calico
+          readOnly: true
+        - mountPath: /opt/cni/bin
```

Comment:
This will shadow the CNI's shipped with `hyperkube` image. Perhaps it could be mounted to a different location and then exposed using `--cni-bin-dir`, as it supports multiple entries.

Reply:
How will this help? And what is the use of the cni that is shipped with hyperkube image?

And this is what we do with bootstrap kubelet as well: https://github.com/kinvolk/lokomotive-kubernetes/blob/71280fe3257e12392054ab689f84acc76464be41/packet/flatcar-linux/kubernetes/cl/controller.yaml.tmpl#L77-L78

Reply:
If we do the same with bootstrap kubelet, I guess it's fine.

This matters, when the CNI configuration deployment process does not actually install the CNI plugins. Then if they are not presented on the host, you end up with a deployed configuration, but `kubelet` can't find any plugins, so it won't work.

But since we don't support such scenario, we're fine here.